src/Components/OATHeader/OATHeader.tsx

```diff
@@ -394,6 +399,48 @@ const OATHeader = ({ elements, dispatch, state }: OATHeaderProps) => {
         onFilesUpload(acceptedFiles);
     }, [acceptedFiles]);
 
+    const onKeyDown = (e) => {
+        if (keyDown.current) {
+            // Prevent keydown from being triggered twice
+            return;
+        }
+        if ((e.key === 'z' && e.ctrlKey) || (e.key === 'z' && e.metaKey)) {
+            if (e.shiftKey) {
+                keyDown.current = true;
+                redoButtonRef.current['_onClick']();
+            } else {
+                keyDown.current = true;
+                undoButtonRef.current['_onClick']();
+            }
+        }
+
+        if ((e.key === 'y' && e.ctrlKey) || (e.key === 'y' && e.metaKey)) {
+            keyDown.current = true;
+            redoButtonRef.current['_onClick']();
+        }
+    };
+
+    useEffect(() => {
+        // Set listener to undo/redo buttons on key press
+        document.addEventListener('keydown', (e) => onKeyDown(e));
+        return () => {
+            document.removeEventListener('keydown', (e) => onKeyDown(e));
```

Comment:
Does this unsubscribe the event handler properly? I.e. we're redefining the handler function here. Does the redefined function === the function we create when we subscribe? To solve this I think we can just pass `onKeyDown` rather than create an inline function.